src/modals/index.tsx

```diff
@@ -2,21 +2,51 @@ import { Dialog, Transition } from '@headlessui/react';
 import { Fragment, ReactNode } from 'react';
 import { ReactComponent as IconTimes } from 'assets/icons/times.svg';
 import { ReactComponent as IconChevron } from 'assets/icons/chevronRight.svg';
+import { ReactComponent as IconBancor } from 'assets/icons/bancor.svg';
+import { CreatePoolModal } from './CreatePoolModal';
+import { DepositDisabledModal } from './DepositDisabledModal';
+import { DurationModal } from './DurationModal';
+import { DepositETHModal } from './DepositETHModal';
+import { SelectPoolModal } from './SelectPoolModal';
+import { UpgradeBntModal } from './UpgradeBntModal';
+import { UpgradeTknModal } from './UpgradeTknModal';
+import { V3BonusesModal } from './V3BonusesModal';
+import { V3ExternalHoldingsModal } from './V3ExternalHoldingsModal';
+import { V3UnstakeModal } from './V3UnstakeModal';
+import { V3WithdrawCancelModal } from './V3WithdrawCancelModal';
 
-interface Props {
-  title?: string;
-  children: JSX.Element;
-  separator?: boolean;
-  titleElement?: ReactNode;
-  setIsOpen: (isOpen: boolean) => void;
-  isOpen: boolean;
-  onBackClick?: Function;
-  showBackButton?: boolean;
-  onClose?: Function;
-  large?: boolean;
+export enum ModalNames {
+  DepositDisabled,
+  DepositETH,
+  Duration,
+  SelectPool,
+  UpgradeBnt,
+  UpgradeTkn,
+  V3Bonuses,
+  V3ExternalHoldings,
+  V3Unstake,
+  V3WithdrawCancel,
 }
 
-export const ModalV3 = ({
+export const Modals = () => {
+  return (
+    <>
+      <CreatePoolModal />
```

Comment:
not sure I understand why are we prerendering part of the modals here like this

Reply:
What caught my attention was that only part of the modals was there.
But to answer your question - you can render <Modals /> to the tree and it will use the redux state to know which children to render. One advantage is that it will allow you to clear stuff upon unmount, in case it will be relevant